### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by chadiel**
Adds the following cmdlets:

Get-AzureProvider
Register-AzureProvider
Unregister-AzureProvider
Get-AzureProviderFeature
Register-AzureProviderFeature

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@chadiel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chadiel). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by amarzavery** [comment]
@chadiel please pull from upstream dev and resolve merge conflicts

**Comment by chadiel** [comment]
I did before sending the CR – am I missing something?

C:\GitHub\chadiel\azure-powershell
$> git merge upstream/dev
Already up-to-date.

C:\GitHub\chadiel\azure-powershell
$> git pull
Already up-to-date.

C:\GitHub\chadiel\azure-powershell
$>

Thanks,


Shadi Majed El Hajj Sleiman
Sr. Software Eng Mgr | Cloud & Enterprise – Modern Apps – Azure Resource Manager (Sparta)
Microsoft Corporation | One Microsoft Way, Redmond WA 98052
shadi.sleiman@microsoft.com<mailto:shadi.slieman@microsoft.com> | Office: +1 (425) 421 8218

**Comment by ogail** [comment]
@chadiel do `git pull upstream dev` that should make it

**Comment by chadiel** [comment]
Done

**Comment by ogail** [comment]
@azuresdkci add to whitelist

File: src/Common/Commands.Common.Storage/Commands.Common.Storage.csproj [hunk 1]
```diff
@@ -166,6 +165,7 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (ogail) [inline src/Common/Commands.Common.Storage/Commands.Common.Storage.csproj]: Remove `app.config`

File: src/Common/Commands.Common.Test/Commands.Common.Test.csproj [hunk 2]
```diff
@@ -209,6 +209,7 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (ogail) [inline src/Common/Commands.Common.Test/Commands.Common.Test.csproj]: Same for all projects

File: src/Common/Commands.ScenarioTest/packages.config [hunk 3]
```diff
@@ -3,8 +3,9 @@
   <package id="Hyak.Common" version="1.0.2" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.0.2" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Common.Extensions" version="0.14.0-preview" targetFramework="net45" />
```
Comment (ogail) [inline src/Common/Commands.ScenarioTest/packages.config]: Please refine from using `Microsoft.Azure.Common.Extensions` its obsolete. Use `Microsoft.Azure.Common.Authorization` instead

Comment (chadiel) [inline src/Common/Commands.ScenarioTest/packages.config]: Will remove - I actually didn't add it and found it was automagically added by VS - maybe there is something broken somewhere?

**Comment by chadiel** [comment]
Updated the CR and addressed all comments - could you please review and merge this pull request?

File: src/Common/Commands.Common.Storage/Commands.Common.Storage.csproj (hunk 1 above)
Comment (chadiel) [inline src/Common/Commands.Common.Storage/Commands.Common.Storage.csproj]: Done

**Comment by azurecla** [comment]
Hi __@chadiel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chadiel). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by chadiel** [comment]
Addressed all CR comments - please review

**Comment by chadiel** [comment]
Seems like the merge job has a bug in it? It fails w/ access denied...

Deleting project workspace... Cannot delete workspace: remote file operation failed: D:/workspace/azure-powershell at hudson.remoting.Channel@1e3d2b3:slave-1: java.nio.file.AccessDeniedException: D:\workspace\azure-powershell\src\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.Tasks.dll
ERROR: Cannot delete workspace: remote file operation failed: D:/workspace/azure-powershell at hudson.remoting.Channel@1e3d2b3:slave-1: java.nio.file.AccessDeniedException: D:\workspace\azure-powershell\src\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.Tasks.dll
[ANALYSIS-COLLECTOR] Skipping publisher since build result is FAILURE
Archiving artifacts

**Comment by ogail** [comment]
@chadiel the PR still have merge conflicts

**Pull request closed without merging**